flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumer.java

```diff
@@ -206,7 +206,13 @@ private void deserializeRecordForCollectionAndUpdateState(UserRecord record)
 		final long approxArrivalTimestamp = record.getApproximateArrivalTimestamp().getTime();
 
 		final T value = deserializer.deserialize(
-			keyBytes, dataBytes, subscribedShard.getStreamName(), record.getSequenceNumber(), approxArrivalTimestamp);
+				keyBytes,
+				dataBytes,
+				record.getPartitionKey(),
+				record.getSequenceNumber(),
+				approxArrivalTimestamp,
+				subscribedShard.getStreamName(),
+				subscribedShard.getShard().getShardId());
```

Comment:
`subscribedShard.getStreamName()` and `subscribedShard.getShard().getShardId()` will be the same for all records collected in the while loop. Probably won't have a big performance difference, but perhaps it would be better to have `final` values for these two to point out this fact?